### DISCUSSION TITLE
Testing `setInterval` or `setTimeout`

### DISCUSSION THREAD

**Discussion by senyai** [microsoft/vscode-discussions#851] [Extension Development QnA] [answered]
Hi! I'm writing scm extension similar to git extension. Can someone give me an idea how to write a test to cover [`this.cleanup()` method](https://github.com/microsoft/vscode/blob/4f7a76b792d38dcf576c831efc3f5c40cd6624e2/extensions/git/src/fileSystemProvider.ts#L53)? I think I want to stub `setInterval` function, but I didn't manage to find a place for code that runs after vscode is loaded but before or when an extension is activated.

**Comment by isidorn** [comment]
What exactly do you want to achieve here? What is the scenario? You should not aim to have your code loaded before your extension is activated

**Reply by senyai** [reply]
I just want to increase code coverage of my extension.

**Comment by connor4312** [comment, marked as answer]
The [fake timers provided by Sinon](https://sinonjs.org/releases/latest/fake-timers/) is what I would use here.

**Reply by senyai** [reply]
The code I want to test runs in the `activate` method. And I'm looking for a place to put fake timers to run before this method.

**Reply by connor4312** [reply]
Extensions during tests follow the standard set of activation events. Unless you activate on * / onStartupFinished, your activate method won't be called until you trigger it with something.
